pkg/providers/amazon/amazon.go

```diff
@@ -0,0 +1,77 @@
+package amazon
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ecr"
+	"github.com/google/go-containerregistry/pkg/authn"
+)
+
+type Provider struct{}
+
+func NewProvider() *Provider {
+	return &Provider{}
+}
+
+func (p Provider) GetAuthKeychain(registryStr string) (authn.Keychain, error) {
+	ecrClient, err := awsRegionalClient(context.TODO(), parseECRDetails(registryStr))
+	if err != nil {
+		return nil, fmt.Errorf("error loading AWS config: %w", err)
+	}
+
+	authTokenOutput, err := ecrClient.GetAuthorizationToken(context.TODO(), &ecr.GetAuthorizationTokenInput{})
```

Comment:
Is `GetAuthorizationToken` called for each image request? 

`AuthorizationData` has the `expiresAt` field, so I assume we can cache the token till this time and avoid flooding AWS with requests.

Reply:
Done